# HG changeset patch
# User Rik <rik@octave.org>
# Date 1375371972 25200
#      Thu Aug 01 08:46:12 2013 -0700
# Node ID 723a08d266d22b2aed39c0dcc641f9759caa8a77
# Parent  7066eec3431c83085feb5ac6ba559faa3cba1f9b
close.m: Add input validation for 2 input args and %!error test blocks.

* scripts/plot/close.m: Add input validation for 2 input args and %!error test
blocks.

diff --git a/scripts/plot/close.m b/scripts/plot/close.m
--- a/scripts/plot/close.m
+++ b/scripts/plot/close.m
@@ -65,22 +65,24 @@ function retval = close (arg1, arg2)
     if (ischar (arg1) && strcmpi (arg1, "all"))
       figs = (get (0, "children"))';
       figs = figs(isfigure (figs));
     elseif (isfigure (arg1))
       figs = arg1;
     elseif (isempty (arg1))
       figs = [];
     else
-      error ('close: expecting argument to be "all" or a figure handle');
+      error ('close: first argument must be "all" or a figure handle');
     endif
   elseif (   ischar (arg1) && strcmpi (arg1, "all")
           && ischar (arg2) && strcmpi (arg2, "hidden"))
     figs = (allchild (0))';
     figs = figs(isfigure (figs));
+  else
+    error ('close: expecting argument to be "all hidden"');
   endif
 
   for h = figs
     __go_execute_callback__ (h, "closerequestfcn");
   endfor
 
   if (nargout > 0)
     retval = 1;
@@ -89,15 +91,21 @@ function retval = close (arg1, arg2)
 endfunction
 
 
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   close (hf);
 %!   objs = findobj ("type", "figure");
-%!   assert (isempty (intersect (objs, hf)));
+%!   assert (! any (objs == hf));
 %! unwind_protect_cleanup
 %!   if (isfigure (hf))
 %!     close (hf);
 %!   endif
 %! end_unwind_protect
 
+%!error close (1,2,3)
+%!error <first argument must be "all" or a figure> close ({"all"})
+%!error <first argument must be "all" or a figure> close ("all_and_more")
+%!error <first argument must be "all" or a figure> close (-1)
+%!error <expecting argument to be "all hidden"> close "all" hid"
+
